### ISSUE TITLE
Change text color of disabled form controls

### ISSUE THREAD

**Issue by julien-deramond** [Orange-OpenSource/Orange-Boosted-Bootstrap#2221]
### Prerequisites

- [X] I have [searched](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/.github/CONTRIBUTING.md)

### Description

A change of text color is needed for disabled form controls for placeholders and/or values. They should use a light gray (`#ccc`).

* The placeholder of a disabled `<select>` form control should be replaced by `#ccc` (the arrow should have the same color)
* The values and placeholders of disabled `<input>` and `<textarea>` should be replaced by `#ccc`
* Labels color should also be gray as described in the [Forms design artboard in the DSM](https://system.design.orange.com/0c1af118d/p/88ab5b-forms/b/599459).
  * Such change could be done associated with `.is-disabled` on the form but what about some disabled form controls within a regular form? **This kind of behavior must be clarified with the designers before starting the development**
* **Be careful** with the global consistency of all disabled form controls that must be checked!

**Comment by MewenLeHo** [comment]
I made some tests in a PR (not yet pushed) and applied `#ccc` on `input text`, `select` and `texarea`:

![screenshot-localhost_9001-2023 09 06-15_59_37](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/assets/92363071/764bfcbc-6640-4d60-b104-5e1f56edb8a6)

Some notes:
- we will need a `disabled` version of the `form-helper` (new svg?)
- even if the `form-helper` is disabled you can display the associated tooltip by hovering the button, you just don't have the hand cursor. Is it a problem?
- we need something else than `#ccc` because of the poor contrast : light grey on light light grey is an accessibility fail (*).

*: a disabled input is an inactive element so yes normally contrast should not be an issue. **But** an inactive element in a form can normally become active when a certain action is performed by the user (validate consent by clicking on a `checkbox` for instance). So one can argue that the user need to know what is inactive and how to activate it if needed and that's why inactive elements in a form must be accessible too. (credit to @Aniort for the demonstration).

**Comment by MewenLeHo** [comment]
Side note: we can add a `.is-disabled` CSS class on the label of the disabled file input.
![screenshot-boosted orange com-2023 09 08-14_27_25](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/assets/92363071/14d7fce4-b3fa-42ba-a8ec-b77ae8b79aff)

**Comment by MewenLeHo** [comment]
Whole page with all (?) possible form elements:

![screenshot-localhost_9001-2023 09 08-15_08_42](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/assets/92363071/3faccf6a-923f-4920-9ce1-fefac39906a2)

@louismaximepiton raised a question about the disabled quantity selector: since it has an editable field, should it turn grey when disabled?

**Comment by MewenLeHo** [comment]
Prototype for a new version of the disabled quantity selector using the same background than the rest of the disabled editable fields:

![screenshot-localhost_9001-2023 09 13-09_55_03](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/assets/92363071/0bb02ff1-a620-444b-8146-f9f13c9b2b4a)

**Comment by julien-deramond** [comment]
The final prototype made by Mewen is in the following closed PR https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/pull/2257 which can be reopened when the development starts.
Live preview: https://deploy-preview-2257--boosted.netlify.app/docs/5.3/examples/form/

**Comment by julien-deramond** [comment]
Note for future selves when we'll get back to this PR.
Dark mode introduced some new concepts, and this PR should probably take that into account. For example:

![Screenshot 2023-10-10 at 11 06 46](https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/assets/17381666/6121074d-d7b5-4eab-acac-034c5e31abce)

**Comment by MaxLardenois** [comment]
Boosted only, won't fix
